app/models/host/base.rb

```diff
@@ -366,7 +375,8 @@ def render_template(template:, **params)
     end
 
     def uptime_seconds
-      self.uptime_fact&.value&.to_i
+      boot_time = self&.reported_data&.boot_time.to_i
+      boot_time > 0 ? Time.zone.now.to_i - boot_time : nil
```

Comment:
what about:
```ruby
boot_time = self&.reported_data&.boot_time
Time.zone.now.to_i - boot_time.to_i unless boot_time.nil? # or if boot_time
```

Reply:
How do we know what timezone the boot time is in? I wonder if this can lead to things like -3 hours uptime in the UI.

Reply:
> what about:

I reverted to `.nil?` and moved `to_i` part lower, I think it's more readable than statement modified at the end of the method

> How do we know what timezone the boot time is in? I wonder if this can lead to things like -3 hours uptime in the UI.

the bootime in /proc/stat is I believe in system time zone, the parsing uses Time.at which assumes boot time in UTC (# of seconds since unix epoch does not have time zone info). When we display it to user, current timezone is used to convert the time in `Time.zone.now`, the same for `boot_time` thanks to active record. When I was testing this, it seemed to work fine.

Reply:
I believe that the `to_i` removes any information about timezone, what makes maybe the `Time.zone` misleading. `Time.now.to_i` should give the same result.